# Microbial Genomics & Metagenomics for Clinical and Public Health Applications

## ðŸ“˜ Course Overview
This hands-on training course introduces participants to the core principles and practical applications of microbial genomics and metagenomics in clinical and public health contexts. Using real-world datasets and reproducible workflows, participants will learn to analyze genomic diversity, evolutionary relationships, and antimicrobial resistance (AMR) profiles of clinically significant pathogens such as:
- _M. tuberculosis_,
- _V. cholerae_, 
- _E. coli_, and
- _S. sonnei_.

The  course explores the role of MGE in resistance spread and equips learners with tools to investigate microbial communities in complex clinical or environmental samples. In addition, participants are given a chance to analyze their data and present preliminary findings to trainers.

## ðŸ“‚ Repository Structure
```bash
â”œâ”€â”€ data/                                   # Sample datasets or links to data sources
â”œâ”€â”€ notebooks/                              # Jupyter or RMarkdown notebooks for analysis
â”œâ”€â”€ workflows/                              # SNextflow pipelines
â”œâ”€â”€ scripts/                                # Utility scripts for preprocessing and analysis
â”œâ”€â”€ docsâ”œâ”€â”€slides/                          # Course slides
        â”œâ”€â”€handouts_reference_material/      # Course handouts and reference materials
â”œâ”€â”€ environment/                            # Information on how the environment was created
â””â”€â”€ README.md                               # This file
```

## ðŸ›  Requirements
Participants or learners:
- Must have a laptop (Linux or MacOS, Git Bash if on Windows) to perform data analysis
- Should have a basic understanding of Linux command-line tools.
- Have access to HPC (recommended)

## Learning Objectives/Outcomes: 
After this workshop participants should be able:
- To **understand** the principles of pathogen genomics and its role in infectious disease surveillance and control.
- To **describe** the workflow of genomic data generation, from sample collection to sequence analysis.
- To **apply** basic bioinformatics tools to analyze pathogen genomic data.
- To **interpret** genomic data for epidemiological insights, including outbreak detection and tracking.
- To **understand** the role of genomics in identifying antimicrobial resistance mechanisms.
